### PR TITLE
API documentation improved and auto generated.

### PR DESCRIPTION
**Pull request by ildyria**
Most of the changes are documentations.

The most important file is:
https://github.com/LycheeOrg/Lychee/blob/scramble-try-2/app/Http/Resources/OpenApi/DataToResponse.php
which basically enables us to parse Spatie Data classes and automatically generate the json responses.

### CODE REVIEW

File: app/Http/Resources/OpenApi/DataToResponse.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	 *
+	 * @return Reference
+	 */
+	public function reference(ObjectType $type): Reference
+	{
+		return new Reference('schemas', $type->name, $this->components);
+	}
+
+	/**
+	 * Given a type we.
```
Comment (d7415) [inline app/Http/Resources/OpenApi/DataToResponse.php]: ...?

File: config/scramble.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	'ui' => [
+		/*
+		 * Define the title of the documentation's website. App name is used when this config is `null`.
+		 */
+		'title' => null,
+
+		/*
+		 * Define the theme of the documentation. Available options are `light` and `dark`.
+		 */
+		'theme' => 'light',
```
Comment (d7415) [inline config/scramble.php:47]: :cry:

**Review by d7415 — APPROVED** [review]

File: config/scramble.php (hunk 2 above)
Comment (ildyria) [reply config/scramble.php:47]: I wish there was a theme slider. :)

File: app/Http/Resources/OpenApi/DataToResponse.php (hunk 1 above)
Comment (ildyria) [reply app/Http/Resources/OpenApi/DataToResponse.php]: fixed [3eda601](https://github.com/LycheeOrg/Lychee/pull/2812/commits/3eda60128c88deff2c98ecea7987edbc6953a4c7)

**Pull request merged**